### ISSUE TITLE
extend ods-core tests - to reflect a real installation qualification

### ISSUE THREAD

**Issue by clemensutschig** [opendevstack/ods-core#646]
**Is your feature request related to a problem? Please describe.**
Today the tests in  https://github.com/opendevstack/ods-core/tree/master/tests only create a project and see if the namespaces and images are there .. that's a great start - but we need more to verify the integrity of an installation.

- [x] Verify all images where built in `ODS_NAMESPACE` (jenkins master / agents / prov app / docgen & webhook proxy) - 
the base functionality for this test is already there (https://github.com/opendevstack/ods-core/blob/master/tests/utils/images.go#L8) - or maybe even better (https://github.com/opendevstack/ods-quickstarters/blob/master/tests/be-typescript-express/jenkinsfile_test.go#L57-L66)
- [x] Verify for the deployments & images etc in `ODS_NAMESPACE` that the are insync to the configuration (in `ods-configuration`) via tailor (maybe pull in @michaelsauter to challenge ideas)
- [x] Verify login for all application deployments configured in `ods-configuration` - nexus, sq
- [x] copy/extend test.sh in sq/nexus to run against the real Installations - and call from /tests

For the existing tests
- [x] verify that in a newly created project - Jenkins and Webhook proxy are running
- [x] create a new project also from provisioning app API ..  (https://www.opendevstack.org/ods-documentation/opendevstack/3.x/provisioning-app/architecture.html#_consuming_rest_apis_in_java)

And all that should create std xunit files & a test report - which is surely somehow already in GO working :)

The good thing is that https://github.com/opendevstack/ods-quickstarters/tree/master/tests already cover provisioning of components /so nothing on this front (yet)

**Comment by clemensutschig** [comment]
@metmajer  @michaelsauter

**Comment by clemensutschig** [comment]
In detail:
1. ODS component Tests (each should be a separate test(method))
1.0 verify that ods-configuration/ods-core.env is available
1.1 verify 'ODS_NAMESPACE' avail in OCP, verify all images are present
1.2 go to 'ODS_NAMESPACE' in ocp and verify - Jenkins & webhook proxy are deployed and running
1.3 go to 'ODS_NAMESPACE' in ocp and verify - Provision app deployed and running
1.4 go to 'ODS_NAMESPACE' in ocp and verify - Nexus & SQ deployed and running
1.5 verify (with tailor) NO diff in any of the above
1.6 verify login to provision app (that one will need to be passed to the tests I guess)
1.7 list all plugins installed in SQ - and verify against a given list (use token in ods-core.env)

**Comment by henrjk** [comment]
I marked up the checkmarks in @clemensutschig initial comment of this issue that were addressed by #673. 

Regarding the in detail of the previous comment the following is done:

1.0 verify that ods-configuration/ods-core.env is available
1.1 verify 'ODS_NAMESPACE' avail in OCP, verify all images are present
1.5 verify (with tailor) NO diff in any of the above
1.7 list all plugins installed in SQ - and verify against a given list (use token in ods-core.env)

1.2 to 1.4 are partially done as the tests now check that the services are  available as seen by openshift and have pods which are running.

However only for nexus/sonarqube functional tests are in place (which invoke the test.sh scripts with --verify).

So we also would like to have similar functional e2e tests for:
- ods webhook proxy
- ods jenkins
- provisioning app
